lib/nostrum/struct/message/reference.ex

```diff
@@ -0,0 +1,40 @@
+defmodule Nostrum.Struct.Message.Reference do
+  @moduledoc """
+  Struct representing a discord message reference.
+  """
+
+  alias Nostrum.{Snowflake, Util}
+
+  defstruct [
+    :message_id,
+    :channel_id,
+    :guild_id
+  ]
+
+  @typedoc "id of the originating message"
+  @type message_id :: Snowflake.t()
+
+  @typedoc "id of the originating message's channel"
+  @type channel_id :: Snowflake.t()
+
+  @typedoc "id of the originating message's guild"
+  @type guild_id :: Snowflake.t()
+
+  @type t :: %__MODULE__{
```

Comment:
How do we feel about actually using the types from these modules. Like `Nostrum.Struct.Message.id` for example?

Reply:
That makes sense to me.